extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -124,7 +140,8 @@ public void build(SmallRyeReactiveMessagingTemplate template, BeanContainerBuild
                         .getIdentifier());
             }
         }
-        template.registerMediators(beanClassToBeanId, beanContainer.getValue());
+        template.registerMediators(beanClassToBeanId, beanContainer.getValue(),
+                emitterFields.stream().map(EmitterBuildItem::getName).collect(Collectors.toList()));
```

Comment:
It seems we don't need `EmitterBuildItem.bean` and `EmitterBuildItem.field` at all?

Reply:
you are right, I just need the name.